From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: OpenWrt <openwrt@openwrt.org>
Date: Wed, 25 Jun 2026 00:00:00 +0000
Subject: arm64: dts: rockchip: CM5 Base Waveshare OLED RST gpio-leds node

Hold GPIO1_B4 (FPC expansion pad 9) high for the Waveshare 1.3" OLED HAT
RST line (40-pin header pin 22). Use gpio-leds (not gpio-hog) with a
companion pinctrl entry in patch 9999.

---
--- a/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-cm5-base.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3588s-orangepi-cm5-base.dts
@@ -41,6 +41,18 @@
 		};
 	};
 
+	waveshare_oled_rst {
+		compatible = "gpio-leds";
+		pinctrl-names = "default";
+		pinctrl-0 = <&waveshare_oled_rst_pin>;
+
+		oled-rst {
+			label = "waveshare-oled-rst";
+			gpios = <&gpio1 RK_PB4 GPIO_ACTIVE_HIGH>;
+			default-state = "on";
+		};
+	};
+
 	gpio-keys {
 		compatible = "gpio-keys";
 		pinctrl-names = "default";
